lib/galaxy/webapps/galaxy/api/groups.py

```diff
@@ -2,61 +2,85 @@
 API operations on Group objects.
 """
 import logging
-from typing import (
-    Any,
-    Dict,
+
+from fastapi import (
+    Body,
+    Path,
 )
 
 from galaxy.managers.context import ProvidesAppContext
 from galaxy.managers.groups import GroupsManager
 from galaxy.schema.fields import EncodedDatabaseIdField
-from galaxy.web import (
-    expose_api,
-    require_admin,
+from galaxy.schema.groups import (
+    GroupCreatePayload,
+    GroupIndexListResponse,
+    GroupShowResponse,
 )
 from galaxy.webapps.galaxy.api import (
-    BaseGalaxyAPIController,
     depends,
+    DependsOnTrans,
+    Router,
 )
 
 log = logging.getLogger(__name__)
 
+router = Router(tags=["groups"])
+
 
-class GroupAPIController(BaseGalaxyAPIController):
-    manager = depends(GroupsManager)
+@router.cbv
+class FastAPIGroups:
+    manager: GroupsManager = depends(GroupsManager)
 
-    @expose_api
-    @require_admin
-    def index(self, trans: ProvidesAppContext, **kwd):
-        """
-        GET /api/groups
-        Displays a collection (list) of groups.
-        """
+    @router.get(
+        "/api/groups",
+        summary="Displays a collection (list) of groups.",
+        require_admin=True,
+    )
+    def index(
+        self,
+        trans: ProvidesAppContext = DependsOnTrans,
+    ) -> GroupIndexListResponse:
+        """GET /api/groups - Displays a collection (list) of groups."""
```

Comment:
These docstrings will appear in the interactive documentation under `http://127.0.0.1:8080/api/docs` once you view the details of that particular route. You should use this to write detailed documentation of the endpoint if needed, but in this case, it is probably fine to just remove it if it's not adding anything to the summary.

The same for the rest of the endpoints.

Reply:
You can completely remove the docstrings line if there is no detailed docs for he endpoint :)